programs/mpl-core/src/plugins/utils.rs

```diff
@@ -415,6 +488,130 @@ pub fn delete_plugin<'a, T: DataBlob>(
             }
         }
 
+        for record in &mut plugin_registry.external_registry {
+            if plugin_offset < record.offset {
+                record.offset -= serialized_plugin.len()
+            }
+        }
+
+        plugin_registry.save(account, new_registry_offset)?;
+
+        resize_or_reallocate_account(account, payer, system_program, new_size)?;
+    } else {
+        return Err(MplCoreError::PluginNotFound.into());
+    }
+
+    Ok(())
+}
+
+/// Remove a plugin from the registry and delete it.
+pub fn delete_external_plugin<'a, T: DataBlob>(
+    plugin_key: &ExternalPluginKey,
+    asset: &T,
+    account: &AccountInfo<'a>,
+    payer: &AccountInfo<'a>,
+    system_program: &AccountInfo<'a>,
+) -> ProgramResult {
+    if asset.get_size() == account.data_len() {
+        return Err(MplCoreError::PluginNotFound.into());
+    }
+
+    //TODO: Bytemuck this.
+    let mut header = PluginHeaderV1::load(account, asset.get_size())?;
+    let mut plugin_registry = PluginRegistryV1::load(account, header.plugin_registry_offset)?;
+
+    let mut result = None;
+    for (i, record) in plugin_registry.external_registry.iter().enumerate() {
+        if record.plugin_type == ExternalPluginType::from(plugin_key)
+            && (match plugin_key {
+                ExternalPluginKey::LifecycleHook(address) | ExternalPluginKey::Oracle(address) => {
+                    let pubkey_offset = record
+                        .offset
+                        .checked_add(1)
+                        .ok_or(MplCoreError::NumericalOverflow)?;
+                    address
+                        == &match Pubkey::deserialize(&mut &account.data.borrow()[pubkey_offset..])
+                        {
+                            Ok(address) => address,
+                            Err(_) => return Err(MplCoreError::DeserializationError.into()),
+                        }
+                }
+                ExternalPluginKey::DataStore(authority) => {
+                    let authority_offset = record
+                        .offset
+                        .checked_add(1)
+                        .ok_or(MplCoreError::NumericalOverflow)?;
+                    authority
+                        == &match Authority::deserialize(
+                            &mut &account.data.borrow()[authority_offset..],
+                        ) {
+                            Ok(authority) => authority,
+                            Err(_) => return Err(MplCoreError::DeserializationError.into()),
+                        }
+                }
+            })
+        {
+            result = Some(i);
+            break;
+        }
+    }
```

Comment:
Could this block be put in a helper function called `find_external_plugin` that returns `(index, record)`?

If it is not in a helper, it seems like this big block to repeat which makes us re-evaluate whether the record should have the whole key to make it easier, but if the helper just has it once maybe we are more OK with the searching process.

Reply:
Yes, done!